patches/node/refactor_allow_embedder_overriding_of_internal_fs_calls.patch

```diff
@@ -53,12 +50,26 @@ index 92554a374d189b3b284546b0e122940e9a175de8..e478c57af873da717c00db73d6719f80
    if (statCache !== null) statCache.set(filename, result);
    return result;
  }
-@@ -257,7 +252,7 @@ function readPackage(requestPath) {
-   const existing = packageJsonCache.get(jsonPath);
-   if (existing !== undefined) return existing;
+diff --git a/lib/internal/modules/package_json_reader.js b/lib/internal/modules/package_json_reader.js
+index 066047b55eb9d82bc1eea66dc54aa68abd116b6b..4fcc65e235692d7d5fe2643571f006c593c4e173 100644
+--- a/lib/internal/modules/package_json_reader.js
++++ b/lib/internal/modules/package_json_reader.js
+@@ -1,7 +1,7 @@
+ 'use strict';
  
--  const json = internalModuleReadJSON(path.toNamespacedPath(jsonPath));
-+  const json = internalFsBinding.internalModuleReadJSON(path.toNamespacedPath(jsonPath));
-   if (json === undefined) {
-     packageJsonCache.set(jsonPath, false);
-     return false;
+ const { SafeMap } = primordials;
+-const { internalModuleReadJSON } = internalBinding('fs');
++const internalFsBinding = internalBinding('fs');
```

Comment:
can `lib/internal/modules/package_json_reader.js` be eliminated ? seems like there is only variable renaming.

Reply:
@deepak1556 this needs to be done for the same reason the initial patch made the same change - if we destructure pre-emptively than it won't use our overridden fs methods at execution time.

Reply:
thanks! I missed the original patch diff.